File: backend/src/main/java/com/devops/incident/model/Report.java
package com.devops.incident.model;

import com.devops.incident.model.enums.ReportFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Report {
    private String id;
    private String incidentId;
    private String title;
    private String content;
    private ReportFormat format;
    private String metadata;
    private Instant generatedAt;
    private Instant createdAt;
}

File: frontend/src/types/report.ts
export type ReportFormat = 'JSON' | 'PDF';

export interface ReportMetadata {
  incidentDuration: string;
  severity: string;
  affectedServices: string[];
  rootCause: string;
  actionsExecuted: number;
  actionsProposed: number;
}

export interface Report {
  id: string;
  incidentId: string;
  title: string;
  content: string;
  format: ReportFormat;
  metadata: ReportMetadata | null;
  generatedAt: string;
  createdAt: string;
}

export interface GenerateReportRequest {
  incidentId: string;
  title: string;
  format: ReportFormat;
}
